stl/inc/xstring

```diff
@@ -2240,8 +2242,14 @@ public:
         return _Result;
     }
 
+    static _CONSTEXPR20 bool _Large_string_engaged(size_type _Strcap) noexcept {
+        // decides whether string is in large mode; has no implication on largeness of _Mysize

```

Comment:
Some added comments in this pr look awkward and need some rewordings😳...

Reply:
I think we may say "determines whether large mode is used for such capacity". Also I think the name of this function should be changed to `_Is_large_capacity`.

Reply:
I believe the name is bad too. However, renaming is not enough, as this function is widely referred to in comments, and there are a lot of other usages of "large" in the class(e.g. `_My_large`, `_Is_large`, `_Right_large`, `//... large mode...`).

I think The usage of `large` is generally distracting, as most of them actually mean allocation. For long-time maintainability, we may need to remove this indirection of meaning from the whole class. However, I think we'd better put this work off -- firstly making an issue about main leftover problems in `basic_string` after this pr being settled, and then making separate prs to do the fix.

Some problems that may need further fix:
1. usage of `"large"`
2. coexistence of `"res"` and `"capacity"` for the same meaning
3. handling of `\0` when doing `copy`. Currently they are hard-code maintained like `traits::copy/move(...,_Size+1)`

Reply:
"Large" is the antonym of "small" - I don't see a problem there. (My opinion shouldn't be taken as definitive - I have lived with this codebase for so long, weird things can be familiar to me.)

Reply:
> I think we may say "determines whether large mode is used for such capacity"

adopted in new push